tests/test_transmission_matrix.py

```diff
@@ -73,6 +80,86 @@ def test_tmatrix_from_abcd_input_types(frequencies, A_list, A_np, A_FreqDat):
     _expect_error_abcd_same_type(A_np, A_FreqDat, A_np, A_np)
     _expect_error_abcd_same_type(A_FreqDat, A_np, A_np, A_np)
 
+def test_tmatrix_from_abcd_broadcasting(frequencies):
+    A = [1, 2, 3]
+    B = [1]
+    C = [1]
+    D = [1]
+    TransmissionMatrix.from_abcd(A, B, C, D, frequencies)
+
+    D = [[1]]
+    TransmissionMatrix.from_abcd(A, B, C, D, frequencies)
+
+    D = [[1], [2]]
+    TransmissionMatrix.from_abcd(A, B, C, D, frequencies)
+
+    D = 1
+    TransmissionMatrix.from_abcd(A, B, C, D, frequencies)
+
+    D = [1, 2]
+    _expect_error_abcd_broadcast(A, B, C, D, frequencies)
```

Comment:
if it is just used here once, I would also paremtrize it, and remove the dedicated method ``_expect_error_abcd_broadcast``